src/testing/factories/juju/CrossModelQuery.types.ts

```diff
@@ -0,0 +1,108 @@
+export type CrossModelQuery = {
+  "application-endpoints"?: Record<string, ApplicationEndpoint>;
+  applications?: Record<string, Application>;
+  controller?: {
+    imestamp?: string;
+    [key: string]: unknown;
+  };
+  machines?: Record<string, Machine>;
+  model?: Model;
+  offers?: Record<string, Offer>;
+  storage?: Record<string, unknown>;
+  [key: string]: unknown;
+};
+
+type ApplicationEndpoint = {
```

Comment:
Most of the types in this file are similar to types we have elsewhere and have the same name. I wonder if it would be helpful to have these all prefixed with `CrossModelQuery...`? It would make them quite long, but would make it clear that these types are only to be used with CrossModelQuery responses.